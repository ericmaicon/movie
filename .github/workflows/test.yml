name: Lint & Test

on: push

jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-node@v2
        with:
          node-version: '14'
      - name: Cache dependencies
        uses: actions/cache@v2.1.4
        id: cache-dependencies
        with:
          path: |
            node_modules
            */*/node_modules
          key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
      - name: Install Dependencies
        if: steps.cache-dependencies.outputs.cache-hit != 'true'
        run: yarn install && yarn bootstrap
      - name: Link
        run: yarn lerna link
      - name: Build deps
        run: yarn build
      - name: Lint
        run: yarn lint
      - name: Run Tests
        run: yarn test
      - uses: codecov/codecov-action@v1
        with:
          token: ${{ secrets.CODECOV_TOKEN }}
          directory: packages/**/coverage
          fail_ci_if_error: true
