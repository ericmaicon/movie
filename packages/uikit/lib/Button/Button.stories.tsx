import React from 'react';

import { Story, Meta } from '@storybook/react';

import Button, { ButtonProps } from './Button';

export default {
  title: 'Form/Button',
  component: Button,
} as Meta;

const Template: Story<ButtonProps> = (args) => <Button {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  children: 'Sign Up',
};

export const Secondary = Template.bind({});
Secondary.args = {
  children: 'Login',
  variant: 'secondary',
};
