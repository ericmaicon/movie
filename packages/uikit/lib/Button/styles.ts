import styled, { css } from 'styled-components';

export interface StyledButtonProps {
  /**
   * Button variant
   */
  variant?: 'primary' | 'secondary';
}

const variantModifier = {
  primary: () => css`
    border: 1px solid ${({ theme }) => theme.colors.red};
    color: ${({ theme }) => theme.colors.red};

    &:hover {
      border: 1px solid ${({ theme }) => theme.colors.black};
      color: ${({ theme }) => theme.colors.black};
    }

    &:active {
      border: 1px solid ${({ theme }) => theme.colors.black};
      color: ${({ theme }) => theme.colors.black};
    }

    &:focus {
      border: 1px solid ${({ theme }) => theme.colors.black};
      color: ${({ theme }) => theme.colors.black};
    }

    &:disabled {
      border: 1px solid ${({ theme }) => theme.colors.gray};
      color: ${({ theme }) => theme.colors.black};
    }
  `,
  secondary: css`
    border: none;
    color: ${({ theme }) => theme.colors.white};

    &:hover {
      color: ${({ theme }) => theme.colors.black};
    }

    &:active {
      color: ${({ theme }) => theme.colors.black};
    }

    &:focus {
      color: ${({ theme }) => theme.colors.black};
    }

    &:disabled {
      color: ${({ theme }) => theme.colors.black};
    }
  `,
};

export const Button = styled.button<StyledButtonProps>`
  background: transparent;
  border-radius: 25px;
  cursor: pointer;
  font-size: 0.2rem;
  padding: 5px 15px;
  ${({ theme }) => theme.typography.text};
  ${({ variant = 'primary' }) => variantModifier[variant]};

  //tablet
  @media (min-width: 768px) and (max-width: 1024px) {
    font-size: 0.5rem;
  }

  //desktop
  @media (min-width: 1025px) {
    font-size: 1rem;
  }
`;
