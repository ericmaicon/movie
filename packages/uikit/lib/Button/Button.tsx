import React, { BaseHTMLAttributes, ReactNode } from 'react';

import * as Styled from './styles';

export interface ButtonProps
  extends BaseHTMLAttributes<HTMLButtonElement>,
    Styled.StyledButtonProps {
  /**
   * Button content.
   */
  children: ReactNode;
}

const Button = ({ children, variant = 'primary', ...rest }: ButtonProps) => {
  return (
    <Styled.Button variant={variant} {...rest}>
      {children}
    </Styled.Button>
  );
};

export default Button;
