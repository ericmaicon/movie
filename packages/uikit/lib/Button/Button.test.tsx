import React from 'react';

import { render } from '~/test/index';

import Button from './Button';

describe('Button', () => {
  it('should render correctly', () => {
    const { getByRole } = render(<Button>Button</Button>);

    const buttonElement = getByRole('button');

    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toHaveTextContent(/^Button$/);
    expect(buttonElement).toMatchSnapshot();
  });

  it('should render a primary button', () => {
    const { getByRole } = render(<Button variant="primary">Button</Button>);

    const buttonElement = getByRole('button');

    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toMatchSnapshot();
  });

  it('should render a secondary button', () => {
    const { getByRole } = render(<Button variant="secondary">Button</Button>);

    const buttonElement = getByRole('button');

    expect(getByRole('button')).toBeInTheDocument();
    expect(buttonElement).toMatchSnapshot();
  });
});
