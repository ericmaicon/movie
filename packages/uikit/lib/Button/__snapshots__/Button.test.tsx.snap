// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a primary button 1`] = `
.c0 {
  background: transparent;
  border-radius: 25px;
  cursor: pointer;
  font-size: 0.2rem;
  padding: 5px 15px;
  font-family: Roboto;
  font-weight: 300;
  font-size: 0.2rem;
  line-height: 1em;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  border: 1px solid #A62218;
  color: #A62218;
}

.c0:hover {
  border: 1px solid #1C1C1C;
  color: #1C1C1C;
}

.c0:active {
  border: 1px solid #1C1C1C;
  color: #1C1C1C;
}

.c0:focus {
  border: 1px solid #1C1C1C;
  color: #1C1C1C;
}

.c0:disabled {
  border: 1px solid #A8B4BE;
  color: #1C1C1C;
}

@media (min-width:768px) and (max-width:1024px) {
  .c0 {
    font-size: 0.5rem;
  }
}

@media (min-width:1025px) {
  .c0 {
    font-size: 1rem;
  }
}

<button
  class="c0"
>
  Button
</button>
`;

exports[`Button should render a secondary button 1`] = `
.c0 {
  background: transparent;
  border-radius: 25px;
  cursor: pointer;
  font-size: 0.2rem;
  padding: 5px 15px;
  font-family: Roboto;
  font-weight: 300;
  font-size: 0.2rem;
  line-height: 1em;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  border: none;
  color: #E9E8E7;
}

.c0:hover {
  color: #1C1C1C;
}

.c0:active {
  color: #1C1C1C;
}

.c0:focus {
  color: #1C1C1C;
}

.c0:disabled {
  color: #1C1C1C;
}

@media (min-width:768px) and (max-width:1024px) {
  .c0 {
    font-size: 0.5rem;
  }
}

@media (min-width:1025px) {
  .c0 {
    font-size: 1rem;
  }
}

<button
  class="c0"
>
  Button
</button>
`;

exports[`Button should render correctly 1`] = `
.c0 {
  background: transparent;
  border-radius: 25px;
  cursor: pointer;
  font-size: 0.2rem;
  padding: 5px 15px;
  font-family: Roboto;
  font-weight: 300;
  font-size: 0.2rem;
  line-height: 1em;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  border: 1px solid #A62218;
  color: #A62218;
}

.c0:hover {
  border: 1px solid #1C1C1C;
  color: #1C1C1C;
}

.c0:active {
  border: 1px solid #1C1C1C;
  color: #1C1C1C;
}

.c0:focus {
  border: 1px solid #1C1C1C;
  color: #1C1C1C;
}

.c0:disabled {
  border: 1px solid #A8B4BE;
  color: #1C1C1C;
}

@media (min-width:768px) and (max-width:1024px) {
  .c0 {
    font-size: 0.5rem;
  }
}

@media (min-width:1025px) {
  .c0 {
    font-size: 1rem;
  }
}

<button
  class="c0"
>
  Button
</button>
`;
